### PR TITLE
Relax location of ta head

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Review by jforissier — COMMENTED** [review]
For commits:
  "Move mv-if-changed() into mk/macros.mk"
  "ta/link.mk: always generate TA dyn_list" [with s/to/two/ in the description]
  `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: scripts/ts_bin_to_c.py [hunk 1]
```diff
@@ -65,6 +66,20 @@ def ta_get_flags(ta_f):
...
+                        # ta_head offset into ELF binary
+                        offs = offs + symbol.entry['st_value']
+                        offs = offs + 20    # Flags offset in ta_head
+                        f.seek(offs)
+                        flags = struct.unpack('<I', f.read(4))[0]
+                        return flags
+
         for s in elffile.iter_sections():
             if get_name(s) == '.ta_head':
                 return struct.unpack('<16x4xI', s.data()[:24])[0]
```
Comment (jforissier) [inline scripts/ts_bin_to_c.py:86]: This deserves a comment: `# Compatibility with older TAs`

**Review by jforissier — COMMENTED** [review]
Reviewing commit "scripts/ts_bin_to_c.py: look for ta_head symbol"

>```
> scripts/ts_bin_to_c.py: look for ta_head symbol
> 
> Legacy TAs has their TA header as at the load address of the TA binary.
>```
True, but `scripts/ts_bin_to_c.py` does not assume it is at the load address, it depends on the fact that it is in its own `.ta_head` section. So I suggest "Legacy TAs have their TA header in a .ta_head section".

One more comment in the patch. With that:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by jforissier — COMMENTED** [review]
For commit "ldelf: look for ta_head symbol":
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by jforissier — COMMENTED** [review]
For "ta: relax location of ta_head":
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jforissier** [comment]
BTW, once `ta_header` is not a header anymore, it could be renamed to `ta_metadata` for example. Although it might be more trouble than we'd like...

**Comment by jenswikl** [comment]
Tags applied.

> BTW, once `ta_header` is not a header anymore, it could be renamed to `ta_metadata` for example. Although it might be more trouble than we'd like...

Yes, I'm afraid so.

**Pull request merged**